main.go

```diff
@@ -30,7 +30,7 @@ func HTTPHandler(h http.Handler) fiber.Handler {
 }
 
 // HTTPMiddleware wraps net/http middleware to fiber middleware
-func HTTPMiddleware(mw func(http.Handler) http.Handler) fiber.Handler {
+func HTTPMiddleware(mw func(http.Handler) http.Handler, contextKeys ...interface{}) fiber.Handler {
```

Comment:
Can you add a unit-test for this?

Reply:
@gaby yes i can :)
Or should I implement the other thing with reflection?
Think that it would just be very messy.
Reflection result:
```
Fields for context.valueCtx
field name: noCopy
value: {}
field name: Request
value: {noCopy:{} Header:{noCopy:{} disableNormalizing:false ... <removed because to long)
field name: Response
value: <removed because to long>
value: []
field name: connID
value: 1
field name: connRequestNum
value: 3
field name: connTime
value: 2023-02-27 13:58:50.8841544 +0100 CET m=+1.859533801
field name: remoteAddr
value: <nil>
field name: time
value: 2023-02-27 13:58:50.9131323 +0100 CET m=+1.888511601
field name: logger
value: {ctx:<nil> logger:<nil>}
field name: s
value: &{noCopy:{} Handler:0x11edf00 ErrorHandler:0x11ede80 HeaderReceived:<nil> ContinueHandler:<nil> Name: Concurrency:262144 ReadBufferSize:4096 WriteBufferSize:4096 ReadTimeout:0s WriteTimeout:0s IdleTimeout:0s MaxConnsPerIP:0 MaxRequestsPerConn:0 MaxKeepaliveDuration:0s MaxIdleWorkerDuration:0s TCPKeepalivePeriod:0s MaxRequestBodySize:4194304 DisableKeepalive:false TCPKeepalive:false ReduceMemoryUsage:false GetOnly:false DisablePreParseMultipartForm:false LogAllErrors:false SecureErrorLogMessage:false DisableHeaderNamesNormalizing:false SleepWhenConcurrencyLimitsExceeded:0s NoDefaultServerHeader:true NoDefaultDate:false NoDefaultContentType:false KeepHijackedConns:false CloseOnShutdown:false StreamRequestBody:false ConnState:<nil> Logger:0x176e7a0 TLSConfig:<nil> FormValueFunc:<nil> nextProtos:map[] concurrency:1 concurrencyCh:0xc00028bec0 perIPConnCounter:{pool:{noCopy:{} local:<nil> localSize:0 victim:<nil> victimSize:0 New:<nil>} lock:{state:0 sema:0} m:map[]} ctxPool:{noCopy:{} local:0xc00017f000 localSize:4 victim:<nil> victimSize:0 New:<nil>} readerPool:{noCopy:{} local:0xc00017f200 localSize:4 victim:<nil> victimSize:0 New:<nil>} writerPool:{noCopy:{} local:0xc00017f800 localSize:4 victim:<nil> victimSize:0 New:<nil>} hijackConnPool:{noCopy:{} local:<nil> localSize:0 victim:<nil> victimSize:0 New:<nil>} ln:[0xc000008f18] idleConns:map[] idleConnsMu:{state:0 sema:0} mu:{state:0 sema:0} open:2 stop:0 done:0xc00028be60}
field name: c
value: &{conn:{fd:0xc0000ac000}}
field name: fbr
value: {c:<nil> ch:0 byteRead:false}
field name: timeoutResponse
value: <nil>
field name: timeoutCh
value: <nil>
field name: timeoutTimer
value: <nil>
field name: hijackHandler
value: <nil>
field name: hijackNoResponse
value: false
field name: formValueFunc
value: <nil>
field name: key
value: 0
field name: val
value: &{request:0xc000128700 sessions:map[websspi:{s:0xc000052460 e:<nil>}]}
field name: key
value: websspi-key-UserInfo
field name: val
value: <removed>
field name: Context
field name: key
field name: val
```

Reply:
@gaby @ReneWerner87 
I mine would be possible.
The field with the key "key" and the real key is always in front of the field with the value. Only here we have fields like: hijackHandler etc..
Means you would still search for "field name: key" and the value to it.
Would be possible in any case. I have worked too little with reflection to know if it would be ok productively.